src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System.Runtime.InteropServices.RuntimeInformation.csproj

```diff
@@ -5,6 +5,14 @@
     <Nullable>enable</Nullable>
     <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Unix;$(NetCoreAppCurrent)-Browser</TargetFrameworks>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(TargetsUnix)' == 'true'">
+    <DefineConstants Condition="'$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'armel'">$(DefineConstants);TARGET_ARM</DefineConstants>
+    <DefineConstants Condition="'$(TargetArchitecture)' == 'arm64'">$(DefineConstants);TARGET_ARM64</DefineConstants>
+    <DefineConstants Condition="'$(TargetArchitecture)' == 'amd64' or '$(TargetArchitecture)' == 'x64'">$(DefineConstants);TARGET_ARM64</DefineConstants>
+    <DefineConstants Condition="'$(TargetArchitecture)' == 'x86'">$(DefineConstants);TARGET_X86</DefineConstants>
+    <DefineConstants Condition="'$(TargetOS)' == 'Browser'">$(DefineConstants);TARGET_WASM</DefineConstants>
+    <DefineConstants Condition="'$(TargetArchitecture)' == 's390x'">$(DefineConstants);TARGET_S390X</DefineConstants>
+  </PropertyGroup>
```

Comment:
We don't rebuild this assembly per architecture.  It's only built once for Windows, once for Unix, and once for Browser wasm.  This won't do the "right thing".

Reply:
Ah you are right. I totally forgot about it. `InferPlatformFromTargetArchitecture` in https://github.com/dotnet/runtime/blob/c59d2005fc61b58365cd877c910642c477bb0a48/Directory.Build.props#L35
is not set for libraries (so SDK sets it to AnyCPU). I have split the property into partials.

Reply:
> I have split the property into partials.

I don't see how this helps.  We don't build this assembly per architecture.

Reply:
Ok, thanks for you patience. So it will be same nuget package with one assembly for Unix (with implementation from the machine it was built on). Then this approach will not work here. 😢

Reply:
Thanks for trying.